vmdb/app/models/miq_enterprise.rb

```diff
@@ -1,17 +1,26 @@
 class MiqEnterprise < ActiveRecord::Base
-  has_many :miq_regions,            -> { MiqRegion.scoped }
-  has_many :ext_management_systems, -> { ExtManagementSystem.scoped }
-  has_many :vms_and_templates,      -> { VmOrTemplate.where.not(:ems_id => nil) }
-  has_many :vms,                    -> { Vm.where.not(:ems_id => nil) }
-  has_many :miq_templates,          -> { MiqTemplate.where.not(:ems_id => nil) }
-  has_many :hosts,                  -> { Host.where.not(:ems_id => nil) }
-  has_many :storages,               -> { Storage.scoped }
-  has_many :policy_events,          -> { PolicyEvent.scoped }
+  has_many :miq_regions,            :class_name => "MiqRegion"
+  has_many :ext_management_systems, :class_name => "ExtManagementSystem"
+  has_many :vms_and_templates,      :class_name => "VmOrTemplate"
+  has_many :vms,                    :class_name => "Vm"
+  has_many :miq_templates,          :class_name => "MiqTemplate"
+  has_many :hosts,                  :class_name => "Host"
+  has_many :storages,               :class_name => "Storage"
+  has_many :policy_events,          :class_name => "PolicyEvent"
 
   has_many :metrics,        :as => :resource  # Destroy will be handled by purger
   has_many :metric_rollups, :as => :resource  # Destroy will be handled by purger
   has_many :vim_performance_states, :as => :resource  # Destroy will be handled by purger
 
+  virtual_has_many :miq_regions,             :class_name => "MiqRegion"
+  virtual_has_many :ext_management_systems,  :class_name => "ExtManagementSystem"
+  virtual_has_many :vms_and_templates,       :class_name => "VmOrTemplate"
+  virtual_has_many :vms,                     :class_name => "Vm"
+  virtual_has_many :miq_templates,           :class_name => "MiqTemplates"
+  virtual_has_many :hosts,                   :class_name => "Host"
+  virtual_has_many :storages,                :class_name => "Storage"
+  virtual_has_many :policy_events,           :class_name => "PolicyEvent"
+
```

Comment:
If you are adding `virtual_has_many`, shouldn't you be deleting `has_many`?  Having both does not make sense.